src/main/io/dataEdition.c

```diff
@@ -0,0 +1,1359 @@
+					tmp.int_value = valueInteger;
+				}
+			}
+			cliSetVar(&valueTable[structureOfPage.editField[3].data], tmp);
+			break;
+		}
+	}
+}
+
+#if (DATA_EDITION_FULL_VERSION)
+static void testAndPositionThatBit(uint8_t testValue, uint32_t bitMask)
+{
+	if (testValue == 1)
+	    featureSet(bitMask);
+	else
+	    featureClear(bitMask);
+}
+
+static void applyNewValuesOnThatPage(void)
+{
+	uint8_t item, index;
+
+    switch (selectedPage){
+	    case FEATURES_PAGE : {
+	        index = (whichOfPages * DISPLAY_LINES_MINUS2) + (selectedLine-1);
+	    	testAndPositionThatBit(structureOfPage.editField[selectedItem].data, (1 << index));
+	        break;
+	    }
+        case MEASURES_PAGE : {
+        	isAlarmSoundOn = structureOfPage.editField[15].data;
+        	masterConfig.telemetryConfig.hottAlarmSoundInterval = structureOfPage.editField[18].data;
+		    break;
+		}
+        case BATTERY_PAGE : {
+	    	testAndPositionThatBit(structureOfPage.editField[3].data, FEATURE_VBAT);
+			masterConfig.batteryConfig.vbatmincellvoltage = structureOfPage.editField[6].data;
+			masterConfig.batteryConfig.vbatmaxcellvoltage = structureOfPage.editField[9].data;
+			masterConfig.batteryConfig.vbatwarningcellvoltage = structureOfPage.editField[12].data;
+			masterConfig.batteryConfig.vbatscale = structureOfPage.editField[15].data;
+
+			batteryWarningVoltage = batteryCellCount * masterConfig.batteryConfig.vbatwarningcellvoltage;
+			batteryCriticalVoltage = batteryCellCount * masterConfig.batteryConfig.vbatmincellvoltage;
+		    break;
+		}
+        case CURRENTMETER_PAGE : {
+	    	testAndPositionThatBit(structureOfPage.editField[3].data, FEATURE_CURRENT_METER);
+        	masterConfig.batteryConfig.currentMeterScale = 10 * (2 * structureOfPage.editField[7].data - 1)
+        			                                       * structureOfPage.editField[6].data;
+        	masterConfig.batteryConfig.currentMeterOffset = 10 * structureOfPage.editField[9].data;
+        	masterConfig.batteryConfig.multiwiiCurrentMeterOutput = structureOfPage.editField[12].data;
+        	masterConfig.batteryConfig.currentMeterType = structureOfPage.editField[15].data;
+		    break;
+		}
+        case ESCMOTORS_PAGE : {
+	    	testAndPositionThatBit(structureOfPage.editField[3].data, FEATURE_MOTOR_STOP);
+	    	testAndPositionThatBit(structureOfPage.editField[6].data, FEATURE_ONESHOT125);
+			masterConfig.escAndServoConfig.minthrottle = (structureOfPage.editField[9].data * 10);
+			masterConfig.escAndServoConfig.servoCenterPulse = (structureOfPage.editField[12].data * 10);
+			masterConfig.escAndServoConfig.maxthrottle = (structureOfPage.editField[15].data * 10);
+			masterConfig.escAndServoConfig.mincommand = (structureOfPage.editField[18].data * 10);
+		    break;
+		}
+	    case PIDCONTROLLER_PAGE : {
+	    	index = getCurrentProfile();
```

Comment:
reusing variables won't improve generated code, but it does decrease readability.
```
int profile = getCurrentProfile();
int item = firstItemOnLine(masterConfig.profile[profile].pidProfile.pidController + 1);
```